src/java/org/apache/cassandra/streaming/StreamingState.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.streaming;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.net.InetSocketAddress;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.db.virtual.SimpleDataSet;
+import org.apache.cassandra.tools.nodetool.formatter.TableBuilder;
+import org.apache.cassandra.utils.Clock;
+import org.assertj.core.util.Throwables;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class StreamingState implements StreamEventHandler
+{
+    public enum State
+    {INIT, START, SUCCESS, FAILURE}
+
+    private final long createdAtMillis = Clock.Global.currentTimeMillis();
+
+    private final UUID id;
+    private final boolean follower;
+    private final StreamOperation operation;
+    // TODO is this changed after init?  looks like it is based off sessions which get added later?
+    private Set<InetSocketAddress> peers;
+    private Sessions sessions = Sessions.EMPTY;
+
+    private State state = State.INIT;
+    private String completeMessage = null;
+
+    private final long[] stateTimesNanos;
+    private volatile long lastUpdatedAtNanos;
+
+    // API for state changes
+    public final Phase phase = new Phase();
+
+    public StreamingState(StreamResultFuture result)
+    {
+        StreamCoordinator coordinator = result.getCoordinator();
+        this.id = result.planId;
+        this.operation = result.getCurrentState().streamOperation;
+        this.follower = coordinator.isFollower();
+        this.peers = coordinator.getPeers();
+        this.stateTimesNanos = new long[State.values().length];
+        stateTimesNanos[0] = Clock.Global.nanoTime();
+    }
+
+    public UUID getId()
```

Comment:
So Cassandra doesn't use the `getXXX` naming. I'd rename this to `id()` instead of `getId()`

Reply:
we are not actually consistent with this, and depends on the authors/reviewers involved